src/checkbox/internal.tsx

```diff
@@ -69,25 +68,20 @@ const InternalCheckbox = React.forwardRef<CheckboxProps.Ref, InternalProps>(
             checked={checked}
             name={name}
             tabIndex={tabIndex}
-            onFocus={onFocus && (() => fireNonCancelableEvent(onFocus))}
-            onBlur={onBlur && (() => fireNonCancelableEvent(onBlur))}
+            onFocus={() => fireNonCancelableEvent(onFocus)}
+            onBlur={() => fireNonCancelableEvent(onBlur)}
             // empty handler to suppress React controllability warning
             onChange={() => {}}
-            onClick={
-              // Using onClick because onChange does not fire in indeterminate state in Internet Explorer and Legacy Edge
-              // https://stackoverflow.com/questions/33523130/ie-does-not-fire-change-event-on-indeterminate-checkbox-when-you-click-on-it
```

Comment:
While the overall behaviour stays the same, the reasoning in this comment does not apply anymore. Since we do not support IE & Legacy Edge, we could use the `onChange` event on the native `input` element. However, we're using `onClick` on the wrapper element to catch any interactions inside the switch (i.e. the native input as well as the label).

Reply:
Hmmm okay, I see now that the `onClick` also moved to the wrapper element.